src/efp.c

```diff
@@ -1991,6 +1991,27 @@ efp_create(void)
 	return efp;
 }
 
+EFP_EXPORT enum efp_result
+efp_set_electron_density_field_fn(struct efp *efp,
+                                  efp_electron_density_field_fn fn)
+{
+    assert(efp);
+
+    efp->get_electron_density_field = fn;
+
+    return EFP_RESULT_SUCCESS;
+}
+
+EFP_EXPORT enum efp_result
+efp_set_electron_density_field_user_data(struct efp *efp, void *user_data)
```

Comment:
fwiw, I was never sure what was supposed to go in this field, so Psi4 doesn't use it.